game-app/game-core/src/main/java/games/strategy/engine/data/gameparser/GameParser.java

```diff
@@ -182,6 +179,7 @@ private GameData parse(final Path xmlFile, final InputStream stream)
 
     parseDelegates(game.getGamePlay().getDelegates());
 
+    variableParser.parseVariables(game.getVariableList());
```

Comment:
There is a side-effect in this call where `parseVariables` caches the variables that were parsed.

WDYT if we made this more functional and `variableParser` returned those parsed variables and did not have any side effects?

Reply:
That's how it was before, see line 206 in the previous diff, but then GameParser had all these functions to operate on these variables. This change moves all that functionality to GameDataVariableParser, so it's all together.

We could refactor things so there's two classes, so there's a wrapper over the returned map. Or another option is we rename GameDataVariableParser to GameDataVariables and have a static parse() function that returns it, this way we don't need two classes.

Reply:
OK, I ended up doing my proposed refactoring above. Please take another look.